### ISSUE TITLE
housenumber not showing up

### ISSUE THREAD

**Issue by rmakkar79** [openstreetmap/Nominatim#1603]
We have nominatim 3.3 , docker based on ubuntu 18
We have imported OSM data followed by 2019 Tiger data using http://nominatim.org/release-docs/3.3.0/admin/Import-and-Update/ 

Everything working fine when we do reverse lookup except in about 20 % of time housenumber is missing.

For example for lat=33.4434341533588&lon=-112.401013845585 :

Our setup with debug flag:
`string(467) "SELECT place_id,parent_place_id,30 as rank_search,ST_LineLocatePoint(linegeo,ST_SetSRID(ST_Point(-112.40101384558,33.443434153359),4326)) as fraction,ST_distance(ST_SetSRID(ST_Point(-112.40101384558,33.443434153359),4326), linegeo) as distance,startnumber,endnumber,interpolationtype FROM location_property_tiger WHERE parent_place_id = 24172969 AND ST_DWithin(ST_SetSRID(ST_Point(-112.40101384558,33.443434153359),4326), linegeo, 0.001) ORDER BY distance ASC limit 1"
object(Nominatim\Result)#6 (5) {
  ["iTable"]=>
  int(0)
  ["iId"]=>
  int(24172969)
  ["iHouseNumber"]=>
  int(-1)
  ["iExactMatches"]=>
  int(0)
  ["iResultRank"]=>
  int(0)`

But on online version -

`string(468) "SELECT place_id,parent_place_id,30 as rank_search,ST_LineLocatePoint(linegeo,ST_SetSRID(ST_Point(-112.40101384558,33.443434153359),4326)) as fraction,ST_distance(ST_SetSRID(ST_Point(-112.40101384558,33.443434153359),4326), linegeo) as distance,startnumber,endnumber,interpolationtype FROM location_property_tiger WHERE parent_place_id = 110568947 AND ST_DWithin(ST_SetSRID(ST_Point(-112.40101384558,33.443434153359),4326), linegeo, 0.001) ORDER BY distance ASC limit 1"
string(23) "found Tiger housenumber"
array(8) {
  ["place_id"]=>
  int(202972410)
  ["parent_place_id"]=>
  int(110568947)
  ["rank_search"]=>
  int(30)
  ["fraction"]=>
  string(17) "0.850441426459758"
  ["distance"]=>
  string(20) "7.39021377073294e-05"`

Any help will be much appreciated.

**Comment by mtmail** [comment]
Can you list a couple of examples in the format
* https://nominatim.openstreetmap.org/reverse.php?format=html&lat=33.4434341533588&lon=-112.401013845585&zoom=21

Are you saying the same queries on your installation of Nominatim return a street address, but no house number? 

In your installation, in the `build/settings/local.php` configuration file, is `CONST_Use_US_Tiger_Data` set to `true`?

**Comment by blucompassai** [comment]
Sure here is one -
https://nominatim.openstreetmap.org/reverse?lat=33.4434341533588&lon=-112.401013845585&format=json

> Are you saying the same queries on your installation of Nominatim return a street address, but no house number?

Yes , that's correct

> In your installation, in the build/settings/local.php configuration file, is CONST_Use_US_Tiger_Data set to true?

Yes, it is set to true. We are getting house number for most of the queries except few like the above. So it's not like that we are not getting it at all.

**Comment by mtmail** [comment]
Can you run the following queries
```sql
select count(*) from location_property_tiger;
=> 33496424
```

```sql
select count(*) from location_property_tiger where postcode='85338';
=> 4611
```

```sql
select * from location_property_tiger where parent_place_id in (select place_id from placex where osm_type='W' and osm_id=151792676) order by startnumber;
=> there should be one start=15857, end=15899, type=odd
```

**Comment by blucompassai** [comment]
Here are results on my local setup :

> nominatim=# select count(*) from location_property_tiger;
  count   
----------
 31317534
(1 row)

> nominatim=# select count(*) from location_property_tiger where postcode='85338';
 count 
-------
     0
(1 row)

> nominatim=# select * from location_property_tiger where parent_place_id in (select place_id from placex where osm_type='W' and osm_id=151792676) order by startnumber;
 linegeo | place_id | partition | parent_place_id | startnumber | endnumber | interpolationtype | postcode 
---------+----------+-----------+-----------------+-------------+-----------+-------------------+----------
(0 rows)

**Comment by mtmail** [comment]
That looks like the import and index didn't finish correctly. The data for this postcode is in `04013.sql`.  Follow http://nominatim.org/release-docs/3.3.0/admin/Import-and-Update/ The `./utils/setup.php --import-tiger-data` first creates a temporary table, later overwrites the `location_property_tiger` table, so you could run in while still serving queries.

**Comment by blucompassai** [comment]
Thanks allot .. so couple of questions:

1. Is there any log i can refer that will tell me what got missed during initial import so i can go back and fix it ?
2. How we know that missing thing is in 04013.sql ? Just in case i need to do for others.
3. Are u suggesting to put only 04013.sql in "data" directory and run import-tiger-data ?
4. As i am running it using docker this is what i did :
- docker exec -it nominatim sudo -u postgres ./src/build/utils/setup.php --import-tiger-data
and then
- docker exec -it nominatim sudo -u postgres ./src/build/utils/setup.php --create-functions --enable-diff-updates --create-partition-functions

Is that correct ?

**Comment by mtmail** [comment]
1. The script output went to STDOUT. I don't think you can see any errors unless the postgresql server logged something related.

2. `04013.sql` contains only about 350,000 lines, you're missing about 2 million. So it must be more files

3. extract all 3233 *.sql files into the directory again. Nominatim will replace the full database table, just adding a single file wouldn't work

4. The command looks alright.

5. great find, that's the issue! Missing files. I corrected that in September in https://github.com/openstreetmap/Nominatim/pull/1474#issuecomment-534767099 but the new files were never uploaded to nominatim.org Use https://downloads.opencagedata.com/public/tiger2019-nominatim-preprocessed.tar.gz until then.

**Comment by lonvia** [comment]
Fixed with #1610.
